controllers/externalsecret_controller.go

```diff
@@ -36,16 +49,113 @@ type ExternalSecretReconciler struct {
 // +kubebuilder:rbac:groups=external-secrets.io,resources=externalsecrets/status,verbs=get;update;patch
 
 func (r *ExternalSecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("externalsecret", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("external-secrets", req.NamespacedName)
+
+	var externalSecret esv1alpha1.ExternalSecret
+
+	err := r.Get(ctx, req.NamespacedName, &externalSecret)
+	if err != nil {
+		log.Error(err, "could not get ExternalSecret")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalSecret.Name,
+			Namespace: externalSecret.Namespace,
+		},
+	}
+
+	_, err = ctrl.CreateOrUpdate(ctx, r.Client, secret, func() error {
+		store, err := r.getStore(ctx, &externalSecret)
+		if err != nil {
+			return fmt.Errorf("could not get store reference from ExternalSecret %q: %w", externalSecret.Name, err)
+		}
+
+		storeProvider, err := schema.GetProvider(store)
+		if err != nil {
+			// TODO: add SecretStore name to the log message
+			return fmt.Errorf("could not get store provider: %w", err)
+		}
+
+		// TODO: Does the * work?
+		// If not, should I change the Provider interface?
+		providerClient, err := storeProvider.New(ctx, *store.GetProvider(), r.Client, req.Namespace)
```

Comment:
It might be more portable to change the storeProvider interface here. Changing the `New` method to accept a GenericStore rather than the concrete typed struct. Will also allow for future versions of the SecretStore resource down the line to all implement the genericStore interface.

```go
New(ctx context.Context, store esv1alpha1.SecretStoreProvider, kube client.Client, namespace string) (Provider, error)
// -> becomes 
New(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (Provider, error)
```

Reply:
Yeah, that works perfectly, thanks ;)